tests/test.rs

```diff
@@ -159,3 +161,24 @@ fn collection_of_enums() {
         }
     );
 }
+
+fn _to_xml<'a>(deserializer: impl Deserializer<'a>) -> Result<String, Box<dyn std::error::Error>> {
+    let mut out = vec![];
+    let mut serializer = serde_xml_rs::ser::Serializer::new(&mut out);
+    serde_transcode::transcode(deserializer, &mut serializer)?;
+    Ok(String::from_utf8(out)?)
+}
+fn _to_json<'a>(deserializer: impl Deserializer<'a>) -> Result<String, Box<dyn std::error::Error>> {
+    let mut out = vec![];
+    let mut serializer = serde_json::ser::Serializer::new(&mut out);
+    serde_transcode::transcode(deserializer, &mut serializer)?;
+    Ok(String::from_utf8(out)?)
+}
+
+#[test]
+fn arbitrary_transcoded_value() {
+    let example_json = r##"{ "Asdasd": ["asdadsda"] }"##;
+    let mut deserializer = serde_json::de::Deserializer::from_str(example_json);
+    let xml = _to_xml(&mut deserializer).expect("failed to transcode json into xml");
+    assert_eq!(r##"<map><field fieldName="Asdasd"><list><item>asdadsda</item></list></field></map>"##.to_string(), xml)
+}
```

Comment:
Again, I would rather not rely on `serde_json` or `serde_transcode` to specify and test the behaviour introduced by this pull request. I believe that the same goal can be accomplished by considering a slightly more general context. This kind of test does have value for checking a specific outcome quickly, but I don't believe that they belong in the final version.

I suggest simply adding a test for the list case, similar to `test_serialize_simple_map`.